diffusion/datasets/image_caption.py

```diff
@@ -65,7 +65,13 @@ def __init__(
         zero_dropped_captions: bool = False,
         **streaming_kwargs,
     ) -> None:
-
+        
+        # Set defaults for vision-friendly streaming args.
+        if 'shuffle_block_size' not in streaming_kwargs:
+            streaming_kwargs['shuffle_block_size'] = 1 << 18
+        if 'shuffle_algo' not in streaming_kwargs:
+            streaming_kwargs['shuffle_algo'] = 'py1s'
```

Comment:
Use the setdefault method for python dicts
```suggestion
        streaming_kwargs.setdefault("shuffle_block_size", 1 << 18)
        streaming_kwargs.setdefault("shuffle_algo", "py1s")
```

Reply:
ooh thanks!